include/Factory.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "macros.h"
+
+class Piece;
+
+template<typename T>
+class Factory {
+
+    static auto &getMap() {
+        static std::map<char, pFnc> m_map;
+        return m_map;
+    }
+
+public:
+    using pFnc = std::unique_ptr<T>(*)();
```

Comment:
pFnc isn't a good name for this function pointer type. Try to think of something more readable that will make the reader understand its purpose. Maybe "creation_func"?